backend/wmg/data/load_cube.py

```diff
@@ -13,37 +13,186 @@
+       B. If the snapshot generation pipeline is run in a "remote dev environment", then the snapshot
+          is stored in s3://<environment_bucket_name>/<remote_stack_name>/<snapshot_id>/
+       C. If validation checks on the snapshot fails, the data artifact will be stored in
+          s3://<environment_bucket_name>/<snapshot_id>_validation_failed_snapshot/
+          (or s3://<environment_bucket_name>/<remote_stack_name>/<snapshot_id>_validation_failed_snapshot/)
+
+    2. The file, s3://<environment_bucket_name>/latest_snapshot_identifer, contains the latest <snapshot_id>
+       that has passed validation. If snapshot generation pipeline is run in a "remote dev environment"
+       then the file will be at path: s3://<environment_bucket_name>/<remote_stack_name>/latest_snapshot_identifer
+
+       NOTE that `latest_snapshot_identifier` will not be updated if validation checks fails on a newly generated
+       data artifact
+
+    3. After each generation of the data artifact and successful validation checks, all but the two latest
+       snapshots are deleted
+
+    Parameters
+    ----------
+    snapshot_source_path: The current source path of the cubes that need to be uploaded to s3
+    snapshot_schema_version: The schema version of the cube data
+    snapshot_id: The snapshot id is a timestamp in seconds
+    is_snapshot_validation_successful: Boolean indicating whether data validation ran sucessfully
+
+    Returns
+    -------
+    snapshot_s3_dest_path: A string that contains the location of the uploaded data
+    """
+    snapshot_id = str(snapshot_id)
+
+    # cleanup files we do not wish to upload
+    shutil.rmtree(f"{snapshot_source_path}/{INTEGRATED_ARRAY_NAME}")
+    shutil.rmtree(f"{snapshot_source_path}/{OBS_ARRAY_NAME}")
+    shutil.rmtree(f"{snapshot_source_path}/{VAR_ARRAY_NAME}")
+
+    snapshot_s3_dest_path = _get_wmg_snapshot_s3_fullpath(
+        snapshot_schema_version, snapshot_id, is_snapshot_validation_successful
+    )
+
+    sync_command = ["aws", "s3", "sync", snapshot_source_path, snapshot_s3_dest_path]
+
+    subprocess.run(sync_command)
+
+    _write_snapshot_metadata(snapshot_schema_version, snapshot_id, is_snapshot_validation_successful)
+
+    return snapshot_s3_dest_path
+
+
+###################################### PRIVATE FUNCTIONS #################################
+
+
+def _get_latest_snapshot_identifier_file_s3_key(snapshot_schema_version: str) -> str:
+    """
+    Return s3 key of latest_snapshot_identifier file
+    """
+    data_root_path = _get_wmg_s3_root_dir_path()
+    _get_wmg_s3_data_schema_dir_path(snapshot_schema_version)
+    file_name = "latest_snapshot_identifier"
+
+    # TODO(prathap) - change to: return f"{data_root_path}/{data_schema_dir_path}/{file_name}"
+    # when working on: https://github.com/chanzuckerberg/single-cell-data-portal/issues/5166
+    return f"{data_root_path}/{file_name}"
+
+
+def _get_latest_snapshot_run_file_s3_key() -> str:
```

Comment:
this function is only used in one place, so i think it would be fine to remove it and move the logic into `_write_latest_snapshot_run_file`

Reply:
agreed. will fix. same for `_get_latest_snapshot_identifier_file_s3_key` as well